### PR TITLE
Fix missing dependency on directly used trilead-api

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: Olivier Lamy <olamy@apache.org>

### Testing done

```[tasklist]
### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: src/main/java/hudson/scm/SubversionSCM.java [hunk 1]
```diff
@@ -3324,6 +3318,8 @@ public FormValidation doCheckLocal(@QueryParameter String value) throws IOExcept
      * Enables trace logging of Ganymed SSH library.
      * <p>
      * Intended to be invoked from Groovy console.
+     * @deprecated
```
Comment (olamy) [inline src/main/java/hudson/scm/SubversionSCM.java:3321]: I  would even remove this (and so dependency on trilead-api) as the class in trilead-api is even already deprecated. 
https://github.com/jenkinsci/trilead-ssh2/blob/d9e5b647ca51e65db86e29a642f85eaee560bd10/src/com/trilead/ssh2/DebugLogger.java#L11

**Comment by MarkEWaite** [comment]
I've confirmed that incremental release 2.17.4-rc1255.4e4a_35243fd5 (from this pull request) resolves the test failure in the plugin bill of materials.  Thanks @olamy !

**Review by MarkEWaite — APPROVED** [review]

**Review by jglick — APPROVED** [review]
@didiez are you still maintaining this? A release is needed. (I have write permissions it seems but would rather not use them. Also #277 would be nice.)

**Comment by timja** [comment]
I'm not maintaining this functionally but I'll help with ecosystem related fixes if there's a working PR provided like in this case.

**Pull request merged**